prow/plugins/mergecommitblocker/mergecommitblocker.go

```diff
@@ -107,15 +107,18 @@ func handle(ghc githubClient, gc git.ClientFactory, cp pruneClient, log *logrus.
 		return err
 	}
 	hasLabel := github.HasLabel(labels.MergeCommits, issueLabels)
-	if hasLabel && !existMergeCommits {
+	hasLabelMergeMethodSquash := github.HasLabel("tide/merge-method-squash", issueLabels)
```

Comment:
Let's not hard code it and add const instead.

Reply:
Adding a constant is actually also problematic. The label value is specified via config so we must respect the value from there: https://github.com/kubernetes/test-infra/blob/86ccd27625d05bfe16376e48eade271b212177f2/prow/config/tide.go#L150-L153

I see that a TODO was added below, but this would be a backwards incompatible change so we can't merge this until the TODO is addressed.
For completeness we should also try to determine if the PR is even in a repo that is served by Tide. If the repo does not use Tide, the presence of the label won't cause the PR to be squashed.

Reply:
Thanks for mentioning this, I agree we should first address TODO and respect the config that we have. @upodroid please also add a test case. 🚀